docs/syntax-attributes-functions-parameters.md

```diff
@@ -6,48 +6,56 @@ This RFC proposes a syntax for function attribute parameters. This is a follow u
 
 ## Design
 
-The following syntax is proposed for attributes with parameters:
+The following syntax is proposed for attributes:
 
 ```ebnf
-attribute = '@' NAME 
-          | '@' NAME '(' ')'
-          | '@' NAME '(' ATOMIC-LITERAL (',' ATOMIC-LITERAL)* ')'
-```
+parameter-table = '{' '}'
+          | '{' parameter (sep parameter)* '}'
 
-The extension proposed to the [Attributes](./syntax-attributes-functions.md) RFC is in the second and third lines which allow zero or more comma-separated literals to be supplied to the attribute. `NAME` should be a valid identifier immediately following `@` without any spaces. The `ATOMIC-LITERAL` category includes values of type `boolean`, `number`, `string`, and the `nil` value. The `nil` value can be be used for "optional" parameters when no valid value exists. As a hypothetical example, consider the `@deprecated(<new-function-name>, <warning-message>)` attribute. Assume a function is being marked deprecated because we are getting rid of the feature entirely and there is no other function replacing it. In that case, we can pass `nil` for the `<new-function-name>` parameter. An alternative would be to allow this attribute to have either one or two parameters. But that still won't help when an attribute can take three parameters and we want to ignore the second. To ensure there is no parsing ambiguity, a `(` following a `NAME`, with or without intervening whitespace, starts a parameter list for that attribute, even if the attribute does not support parameters. The parser will only check for syntacting correctness. The implemention will raise an error when it uses the attribute and finds that an invalid number or type of arguments were supplied. The implemention will treat an attribute with an empty parameter list equivalent to one without parameters. This means that `@native()` is the same as `@native`. This gives us forward compatibility if we decide to support attributes on parenthesized expressions in the future. Consider the following hypothetical example with a `@my_attribute` attribute, which does not take any parameters, applied to a parenthesized expression.
+sep = ',' | ';'
 
-```lua
-@my_attribute
-(f().x)
-```
+parameter = literal 
+          | NAME '=' literal
 
-Since an opening parenthesis starts a parameter list for the attribute, the expression `(f().x)` will be treated as a parameter list of `@my_attribute`. This will lead to a syntax error since attribute parameter can only be `ATOMIC-LITERAL` and `f().x` is not one. This ambiguity can be resolved by adding `()` to `@my_attribute` as follows:
+literal = BOOLEAN | NUMBER | STRING | NIL | parameter-table
 
-```lua
-@my_attribute()
-(f().x)
+list-attribute = NAME parameter-table
+               | NAME
+
+attribute = '@' NAME 
+          | '@[' list-attribute (',' list-attribute)* ']'
```

Comment:
Ditto.
```suggestion
          | '@[' list-attribute {',' list-attribute} ']'
```

Reply:
The `ebnf` syntax has been reworked entirely and this issue is fixed.